toolkit/scripts/containerized-build/resources/setup_functions.sh

```diff
@@ -0,0 +1,146 @@
+    echo -e "* \n* Local repo information:"
+    if [[ "${IS_REPO_ENABLED}" == "true" ]]; then
+        echo -e "*\tLocal repo is enabled. Package dependencies will be installed from $RPMS_DIR, /repo and upstream server"
+    else
+        echo -e "*\tLocal repo is not enabled. Package dependencies will be installed from upstream server"
+    fi
+    echo "******************************************************************************************"
+}
+
+# Refresh repo cache with newly built RPM, use Mariner specific DEFINES
+rpmbuild() {
+    local args=("$@")
+    command "$FUNCNAME" "${DEFINES[@]}" "${args[@]}"
+    if [[ ${IS_REPO_ENABLED} = true ]] ; then
+        refresh_local_repo
+        tdnf makecache
+    fi
+}
+
+# Refresh metadata for local RPMs' repo
+refresh_local_repo() {
+    echo "-------- refreshing the local repo ---------"
+    pushd $RPMS_DIR
+    createrepo .
+    popd
+}
+
+# Satisfy dependencies from local RPMs
+enable_local_repo() {
+    echo "-------- enabling local repo ---------"
+    IS_REPO_ENABLED=true
+    tdnf install -y createrepo
+    mv /etc/yum.repos.d/local_repo.disabled_repo /etc/yum.repos.d/local_repo.repo
+    pushd /repo
+    createrepo .
+    popd
+    mkdir -p $RPMS_DIR
+    pushd $RPMS_DIR
+    createrepo .
+    popd
+    echo "-------- the local repo is enabled ---------"
+    echo "--- Package dependencies will be installed from $RPMS_DIR, /repo and upstream server ---"
+    #echo "You can install the following packages from it:"
+    #tdnf repoquery --repoid=local_build_repo 2>/dev/null
+}
+
+# Update dependency graph using build tools
+update_specs_metadata() {
+    # update specs.json
+    /mariner_setup_dir/specreader --dir=$SPECS_DIR  --srpm-dir="/mnt/INTERMEDIATE_SRPMS/" --output=/mariner_setup_dir/specs.json --dist-tag="containerized" --rpm-dir="/mnt/RPMS/"
+    # update graph.dot
+    /mariner_setup_dir/grapher --input=/mariner_setup_dir/specs.json --output=/mariner_setup_dir/graph.dot
+}
+
+# Install package dependencies using depsearch tool
+install_dependencies_depsearch() {
+    local PKG=("$@")
+    if [ -z "$PKG" ]; then echo "Please provide pkg name"; return; fi
+    echo "-------- installing dependencies ---------"
+    dep_list=$(get_pkg_dependency $PKG)
+    for dependency in $dep_list
+    do
+        tdnf install -y $dependency 2>&1
+    done
+}
+
+# Get dependencies of a package using depsearch tool
+get_pkg_dependency() {
+    local PKG=("$@")
+    if [ -z "$PKG" ]; then echo "Please provide pkg name"; return; fi
+    /mariner_setup_dir/depsearch --input=/mariner_setup_dir/graph.dot  --packages=$PKG --reverse
+}
+
+# Install package dependencies listed as BuildRequires in spec
+install_dependencies() {
+    local PKG=("$@")
+    if [ -z "$PKG" ]; then echo "Please provide pkg name"; return; fi
+    echo "-------- installing build dependencies ---------"
+    spec_file=$SPECS_DIR/$PKG/$PKG.spec
+    dep_list=$(grep "BuildRequires:" $spec_file | cut -d ':' -f 2)
```

Comment:
I have added Mariner specific DEFINES and MACROS (setup_functions.sh #13) to be used when invoking rpmspec and rpmbuild. The case you mentioned here is covered, but if there are other scenarios of fail, we can fix those

Reply:
The `grep` part will not correctly list the right build-requires for many packages with %check disabled. It will install too many dependencies, which can affect the behavior of the package build (it may optionally enable additional features we wouldn't expect)